test/PointerCasts/phi-from-gep.ll

```diff
@@ -23,5 +23,50 @@ block:
 end:
 ; CHECK: phi ptr addrspace(1) [ [[gep]], %entry ], [ [[gepblock]], %block ]
   %phi = phi ptr addrspace(1) [ %0, %entry ], [ %1, %block ]
+  %gep = getelementptr i8, ptr addrspace(1) %phi, i32 7
+  ret void
+}
+
+define spir_kernel void @test2(ptr addrspace(1) %a, i32 %i) {
+entry:
+; CHECK: entry
+; CHECK-NEXT: [[shl0:%[^ ]+]] = shl i32 %i, 2
+; CHECK-NEXT: [[shl1:%[^ ]+]] = shl i32 [[shl0]], 1
+; CHECK-NEXT: [[add:%[^ ]+]] = add i32 [[shl1]], 4
+; CHECK-NEXT: [[gep:%[^ ]+]] = getelementptr i16, ptr addrspace(1) %a, i32 [[add]]
+  %0 = getelementptr <4 x i32>, ptr addrspace(1) %a, i32 %i, i32 2
+  br label %end
+
+block:
+; CHECK: block
+; CHECK-NEXT: [[lshr:%[^ ]+]] = lshr i32 %i, 1
+; CHECK-NEXT: [[gepblock:%[^ ]+]] = getelementptr i16, ptr addrspace(1) %a, i32 [[lshr]]
+  %1 = getelementptr i8, ptr addrspace(1) %a, i32 %i
+  br label %end
+
+end:
+; CHECK: phi ptr addrspace(1) [ [[gep]], %entry ], [ [[gepblock]], %block ]
+  %phi = phi ptr addrspace(1) [ %0, %entry ], [ %1, %block ]
+  %gep = getelementptr i16, ptr addrspace(1) %phi, i32 7
+  ret void
+}
+
+define spir_kernel void @test3(ptr addrspace(1) %a, i32 %i) {
+entry:
+; CHECK: entry
+; CHECK-NEXT: [[gep:%[^ ]+]] = getelementptr <4 x i32>, ptr addrspace(1) %a, i32 %i, i32 0
+  %0 = getelementptr <4 x i32>, ptr addrspace(1) %a, i32 %i
+  br label %end
+
+block:
+; CHECK: block
+; CHECK-NEXT: [[gepblock:%[^ ]+]] = getelementptr <4 x i32>, ptr addrspace(1) %a, i32 0, i32 %i
```

Comment:
This gep seems wrong. SPIR-V requires  that the index into the vector be in bounds. What happens if `i` is greater than 4? I think it should be more like:
```
%div = sdiv i32 %i, 4
%mod = smod i32 %i, 4
%gepblock = getelementptr <4 x i32>, ptr addrspace(1) %a, i32 %div, i32 %mod
```

Is this adjusted later? If so, can you add a two pass test that shows it?

Reply:
This test is showing an issue that has nothing to do with this current PR. I have created #1243 to track the issue.
I have also changed this test to avoid having this wrong pattern.